### PR TITLE
Fix the crash of command fg

### PR DESCRIPTION
**Pull request by PeiweiHu**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Reproduce the crash:
 - rizin /bin/ls
 - fg

**Test plan**

CI is green
...

**Closing issues**

...

**Addition**

```
AddressSanitizer:DEADLYSIGNAL
=================================================================
==55838==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000000 (pc 0x7f2e05d607fd bp 0x7ffef7c8e3e0 sp 0x7ffef7c8db78 T0)
==55838==The signal is caused by a READ memory access.
==55838==Hint: address points to the zero page.
    #0 0x7f2e05d607fd  (/usr/lib/libc.so.6+0x1607fd)
    #1 0x7f2e07447351 in __interceptor_strlen /usr/src/debug/gcc/libsanitizer/sanitizer_common/sanitizer_common_interceptors.inc:387
    #2 0x7f2e06da696b in calcsize_key ../subprojects/sdb/src/ht_inc.c:36
    #3 0x7f2e06da696b in ht_pp_find_kv ../subprojects/sdb/src/ht_inc.c:309
    #4 0x7f2e06da6f8d in ht_pp_find ../subprojects/sdb/src/ht_inc.c:328
    #5 0x7f2e0666d8cd in rz_flag_get ../librz/flag/flag.c:312
    #6 0x7f2dfdd1da13 in __printRecursive ../librz/core/cmd/cmd_flag.c:141
    #7 0x7f2dfdd208ed in rz_flag_graph_handler ../librz/core/cmd/cmd_flag.c:302
    #8 0x7f2dfdd0cb7f in argv_call_cb ../librz/core/cmd/cmd_api.c:747
    #9 0x7f2dfdd0cb7f in call_cd ../librz/core/cmd/cmd_api.c:784
    #10 0x7f2dfdd0cb7f in rz_cmd_call_parsed_args ../librz/core/cmd/cmd_api.c:802
    #11 0x7f2dfdcab908 in handle_ts_arged_stmt_internal ../librz/core/cmd/cmd.c:3692
    #12 0x7f2dfdcab908 in handle_ts_arged_stmt ../librz/core/cmd/cmd.c:3639
    #13 0x7f2dfdba31db in handle_ts_stmt ../librz/core/cmd/cmd.c:5116
    #14 0x7f2dfdc48a2a in handle_ts_statements_internal ../librz/core/cmd/cmd.c:5173
    #15 0x7f2dfdc48a2a in handle_ts_statements ../librz/core/cmd/cmd.c:5138
    #16 0x7f2dfdc49f8a in core_cmd_tsrzcmd ../librz/core/cmd/cmd.c:5281
    #17 0x7f2dfdc4a141 in rz_core_cmd ../librz/core/cmd/cmd.c:5329
    #18 0x7f2dfda278ec in rz_core_prompt_exec ../librz/core/core.c:2828
    #19 0x7f2dfda293c5 in rz_core_prompt_loop ../librz/core/core.c:2698
    #20 0x7f2e06723674 in rz_main_rizin ../librz/main/rizin.c:1410
    #21 0x7f2e05c2928f  (/usr/lib/libc.so.6+0x2928f)
    #22 0x7f2e05c29349 in __libc_start_main (/usr/lib/libc.so.6+0x29349)
    #23 0x561842f27964 in _start (/usr/local/bin/rizin+0x2964)

AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV (/usr/lib/libc.so.6+0x1607fd) 
==55838==ABORTING

```

### CODE REVIEW

File: librz/core/cmd/cmd_flag.c [hunk 1]
```diff
@@ -138,7 +138,7 @@ static void __printRecursive(RzCore *core, RzList *flags, const char *name, RzOu
 		rz_cons_printf("agn root\n");
 		return;
 	}
-	if (rz_flag_get(core->flags, name)) {
+	if (RZ_STR_ISEMPTY(name) || rz_flag_get(core->flags, name)) {
```
Comment (wargio) [inline librz/core/cmd/cmd_flag.c]: yeah, that flag check looks wrong.

File: test/db/cmd/cmd_flags [hunk 2]
```diff
@@ -652,3 +652,40 @@ EXPECT=<<EOF
...
+            terp
+            it
+        ..note.
+               ABI_tag
+               gnu.build_id
+        ..g
+           nu.
+              hash
+              version
+           nu.version
```
Comment (ret2libc) [inline test/db/cmd/cmd_flags:672]: this seems wrong

File: test/db/cmd/cmd_flags [hunk 3]
```diff
@@ -652,3 +652,40 @@ EXPECT=<<EOF
...
+           ot
+        ..d
+           yn
+             s
+              ym
+              tr
+             str
+             amic
+           ynstr
```
Comment (ret2libc) [inline test/db/cmd/cmd_flags:681]: also this seems wrong

File: librz/core/cmd/cmd_flag.c [hunk 4]
```diff
@@ -299,7 +298,11 @@ RZ_IPI RzCmdStatus rz_flag_local_list_all_handler(RzCore *core, int argc, const
 RZ_IPI RzCmdStatus rz_flag_graph_handler(RzCore *core, int argc, const char **argv, RzCmdStateOutput *state) {
 	RzList *flags = rz_list_newf(NULL);
 	rz_flag_foreach_space(core->flags, rz_flag_space_cur(core->flags), listFlag, flags);
-	__printRecursive(core, flags, argv[1], state->mode, 0);
+	if (!argv[1]) {
+		__printRecursive(core, flags, "\0", state->mode, 0);
```
Comment (ret2libc) [inline librz/core/cmd/cmd_flag.c]:
```suggestion
		__printRecursive(core, flags, "", state->mode, 0);
```

**Review by ret2libc — APPROVED** [review]
Ok to merge, but the usefulness of this command is a bit limited at this time I'd say. It can probably be rewritten in an easier way rather than what it is right now.

**Review by ret2libc — APPROVED** [review]

**Pull request merged**